google/cloud/spanner_v1/snapshot.py

```diff
@@ -129,7 +142,14 @@ def _restart_on_unavailable(
                 if transaction is not None:
                     transaction_selector = transaction._make_txn_selector()
                 request.transaction = transaction_selector
-                iterator = method(request=request)
+                nth_request = next_nth_request()
```

Comment:
This is a retry, so it should use the same nth_request, but a higher attempt number.

Reply:
done, thanks!